proto/v1alpha1/state.proto

```diff
@@ -10,12 +10,15 @@ enum EventType {
     CREATED = 0;
     UPDATED = 1;
     DESTROYED = 2;
+    BOOTSTRAPPED = 3;
+    ERRORED = 4;
```

Comment:
I was thinking, maybe we could keep the whole thing backward-compatible by doing the following:
- Do not introduce these 2 new types
- For errors, only use the new error field on the event - it would indicate an error if it is present
- For bootstrapped, use `CREATED` but add additional (e.g. enum) field to the event to distinguish between a regular create and a bootstrap.

Reply:
I feel it's a bit of a hack (I feel cleaner APIs are a win in the long run), but the most important thing for me here is that it won't help: the current COSI protobuf client always assumes it can unmarshal `event.Resource`, so to make it "compatible" we'd have to introduce something in that field which can be unmarshalled back. 

My concern with compatibility was the following:

* I run talosctl 1.2.0 against Talos 1.3.0
* `talosctl` is compiled against old COSI library, so it will panic if it receives an event with `nil` `event.Resource`

Doing the opposite is fine, but the client won't receive new Bootstrapped event, so we need to wait before relying on its presence.

Reply:
Aha alright, all make sense then. I agree that cleaner APIs are better.